var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -50,6 +50,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     version("3.0.00", sha256="c00613d0194a4fbd0726719bbed8b0404ed06275f310189b3493f5739042a92b")
 
     depends_on("cmake@3.16:", type="build")
+    conflicts("cmake@3.28:", when="@:4.2.01 +cuda")
```

Comment:
I'll submit this separately, since it might be needed also in #43349

Reply:
From the GL CI, this change is not required for CMake 3.29
https://gitlab.spack.io/spack/spack/-/jobs/10789546

Reply:
If it is not required, the directive will be wrong as soon as 3.29 gets merged. Do we know why CMake 3.29 seems to work whereas CMake 3.28 does not?

Reply:
I'm not 100% but likely https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9243 this is why 3.29 is fine and 3.28 is not. Updated the conflict.